src/segment-loader.js

```diff
@@ -1255,40 +1262,45 @@ export default class SegmentLoader extends videojs.EventTarget {
       // we're downloading a single rate (and don't know what that rate is)
       // so we just use the seconds in the buffer to calculate the new buffer
       // sizes.
-      maxForwardBuffer = Math.max((forwardBuffer / configTotalBuffer) * secondsInBuffer,
+      newForwardBufferGoal = Math.max((forwardBufferGoal / totalBufferGoal) * secondsInBuffer,
         this.playlist_.targetDuration);
-      maxReverseBuffer = secondsInBuffer - maxForwardBuffer;
+      newBackBufferGoal = secondsInBuffer - newForwardBufferGoal;
     } else {
       // since we potentially have more than one rate in the buffer, we use
       // the amount of bytes in the buffer to calculate the upper limit on
       // the equivalent time at the current rate
       let upperTimeLimit = (bytesInBuffer * 8) / this.playlist_.attributes.BANDWIDTH;
 
-      maxForwardBuffer = (forwardBuffer / configTotalBuffer) * upperTimeLimit;
-      maxReverseBuffer = (reverseBuffer / configTotalBuffer) * upperTimeLimit;
+      newForwardBufferGoal = (forwardBufferGoal / totalBufferGoal) * upperTimeLimit;
+      newBackBufferGoal = (backBufferGoal / totalBufferGoal) * upperTimeLimit;
     }
 
-    maxForwardBuffer = Math.min(forwardBuffer, maxForwardBuffer);
-    maxReverseBuffer = Math.min(reverseBuffer, maxReverseBuffer);
+    newForwardBufferGoal = Math.min(forwardBufferGoal, newForwardBufferGoal);
+    newBackBufferGoal = Math.min(backBufferGoal, newBackBufferGoal);
 
-    loggerMsg = '\n\tbytes in buffer: ' + bytesInBuffer +
-      ', seconds in buffer: ' + secondsInBuffer +
-      '\n\tforward buffer: ' + forwardBuffer + ' => ' + maxForwardBuffer +
-      '\n\treverse buffer: ' + reverseBuffer + ' => ' + maxReverseBuffer;
+    loggerMsg =
+      `\n\tbytes in buffer: ${bytesInBuffer}, seconds in buffer: ${secondsInBuffer}` +
+      `\n\tforward buffer: ${forwardBufferGoal} => ${newForwardBufferGoal}` +
+      `\n\treverse buffer: ${backBufferGoal} => ${newBackBufferGoal}`;
 
     if (setByteLimit) {
-      loggerMsg += '\n\tbyte limit: ' + this.maxBytes_ + ' => ' + bytesInBuffer;
+      loggerMsg += `\n\tbyte limit: ${this.maxBytes_} => ${bytesInBuffer}`;
       this.maxBytes_ = Math.min(this.maxBytes_, bytesInBuffer);
     }
     this.logger_('adjustBuffers_', loggerMsg);
 
-    this.goalBufferLength_(maxForwardBuffer);
-    this.backBufferLength_(maxReverseBuffer);
+    this.goalBufferLength_(newForwardBufferGoal);
+    this.backBufferLength_(newBackBufferGoal);
   }
 
   /**
-   * handler to run when a call to appendBuffer fails because the source
-   * buffer can't hold any more information
+   * Handle a call to appendBuffer that failed
+   *
+   * Depending on what caused the QuotaExceededError, one of two things will happen.
+   * If it happened when appending the first segment of a stream, it will be split
+   * into smaller pieces and appended immediately. If it happened in the regular
+   * course of playing a feed, it will wait until there's sufficient room in the
```

Comment:
Was there are reason for using different strategies here? If the error happens after the first segment, couldn't we remove from the back buffer and then append small pieces of the next segment instead of waiting for enough space to append the whole segment?